### PR TITLE
Provide tree deps in EDN format for display/parsing by other tools

### PR DESCRIPTION
**Pull request by kenrestivo-stem**
Support outputting tree data as EDN for parsing by other programs.

### CODE REVIEW

File: src/leiningen/deps.clj [hunk 1]
```diff
@@ -123,7 +130,10 @@ force them to be updated, use `lein -U $TASK`."
                               dependencies-key
                               managed-dependencies-key
                               project)]
-               (walk-deps hierarchy print-dep))
+               (case command
+                 ":tree" (walk-deps hierarchy print-dep)
+                 ":tree-data"  (binding [*print-length* 10000 *print-level* 10000]
```
Comment (danielcompton) [inline src/leiningen/deps.clj]: Looking at the failed tests, it seems like you also need to handle `:plugin-tree` here?

**Comment by danielcompton** [comment]
It looks like this is the successor to #2255.

File: src/leiningen/deps.clj (hunk 1 above)
Comment (danielcompton) [inline src/leiningen/deps.clj]: Also, that means `:plugin-tree-data` should probably also be provided?

Comment (kenrestivo-stem) [inline src/leiningen/deps.clj]: OK, will do.

**Comment by kenrestivo-stem** [comment]
Should be ready to go, thanks.

**Comment by technomancy** [comment]
Looks great; thanks!

**Pull request merged**